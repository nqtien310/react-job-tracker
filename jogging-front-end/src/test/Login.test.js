import React from 'react';
import renderer from 'react-test-renderer'
import LoginForm from '../Login'
import RenderComponent from './RenderComponent'

it('Login Form', () => {
  const component = renderer.create(
    <RenderComponent>
      <LoginForm/>
    </RenderComponent>
  )

  expect(component).toMatchSnapshot()
});

