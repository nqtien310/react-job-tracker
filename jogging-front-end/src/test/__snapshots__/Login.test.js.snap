// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login Form 1`] = `
<div>
  <nav
    className="navbar navbar-toggleable-md navbar-inverse bg-inverse"
  >
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className="navbar-brand"
      href="#"
    >
      <i
        className="fa fa-home"
      />
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    id="body"
  >
    <div
      id="login-form"
    >
      <form
        onSubmit={[Function]}
      >
        <div />
        <div>
           
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={true}
                className="form-control"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="password"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
           
        </div>
        <div
          className="form-group row"
        >
          <div
            className="offset-sm-3 col-sm-9"
          >
            Do not have account yet ?
            <a
              href="/register"
              onClick={[Function]}
            >
               register here 
            </a>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <div
            className="offset-sm-3 col-sm-9"
          >
            <button
              className="btn btn-primary"
              type="Submit"
            >
               
              Login
               
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders error message when login failed 1`] = `
<div>
  <nav
    className="navbar navbar-toggleable-md navbar-inverse bg-inverse"
  >
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <a
      className="navbar-brand"
      href="#"
    >
      <i
        className="fa fa-home"
      />
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item active"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    id="body"
  >
    <div
      id="login-form"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="alert alert-danger"
        >
          <ul>
            <li>
               
              Invalid username or password
               
            </li>
          </ul>
        </div>
        <div>
           
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="email"
            >
              Email
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={true}
                className="form-control"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group row"
          >
            <label
              className="col-sm-3 col-form-label"
              htmlFor="password"
            >
              Password
            </label>
            <div
              className="col-sm-9"
            >
              <input
                autoFocus={false}
                className="form-control"
                id="password"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
            </div>
          </div>
           
        </div>
        <div
          className="form-group row"
        >
          <div
            className="offset-sm-3 col-sm-9"
          >
            Do not have account yet ?
            <a
              href="/register"
              onClick={[Function]}
            >
               register here 
            </a>
          </div>
        </div>
        <div
          className="form-group row"
        >
          <div
            className="offset-sm-3 col-sm-9"
          >
            <button
              className="btn btn-primary"
              type="Submit"
            >
               
              Login
               
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
